src/main/java/de/uhd/ifi/se/decision/management/jira/rest/DashboardRest.java

```diff
@@ -90,40 +91,34 @@ public Response getRationaleCompleteness(@Context HttpServletRequest request, Fi
 	@POST
 	@Produces({ MediaType.APPLICATION_JSON })
 	public Response getRationaleCoverage(@Context HttpServletRequest request, FilterSettings filterSettings,
-			@QueryParam("issueType") String issueType) {
-		if (request == null || filterSettings == null || issueType == null) {
+			@QueryParam("sourceKnowledgeTypes") String sourceKnowledgeTypesString) {
+		if (request == null || filterSettings == null || sourceKnowledgeTypesString == null) {
 			return Response.status(Status.BAD_REQUEST).entity(ImmutableMap.of("error", "There is no project selected"))
 					.build();
 		}
 
+		Set<String> sourceKnowledgeTypes;
+		if (sourceKnowledgeTypesString.isEmpty()) {
+			sourceKnowledgeTypes = new DecisionKnowledgeProject(filterSettings.getProjectKey()).getNamesOfKnowledgeTypes();
+		} else {
+			sourceKnowledgeTypes = new HashSet<>(Arrays.asList(sourceKnowledgeTypesString.split(",")));
+		}
+
 		ApplicationUser user = AuthenticationManager.getUser(request);
 
 		Map<String, Object> metrics = new LinkedHashMap<>();
 
 		RationaleCoverageCalculator rationaleCoverageCalculator = new RationaleCoverageCalculator(user, filterSettings);
 
-		// TODO Get rid of this if-else block, only work with knowledge elements
-		if (issueType.equals("Code")) {
-			metrics.put("issuesPerJiraIssue",
-					rationaleCoverageCalculator.getNumberOfDecisionKnowledgeElementsForCodeFiles(KnowledgeType.ISSUE));
-			metrics.put("decisionsPerJiraIssue", rationaleCoverageCalculator
-					.getNumberOfDecisionKnowledgeElementsForCodeFiles(KnowledgeType.DECISION));
-			metrics.put("decisionDocumentedForSelectedJiraIssue",
-					rationaleCoverageCalculator.getCodeFilesWithNeighborsOfOtherType(KnowledgeType.ISSUE));
-			metrics.put("issueDocumentedForSelectedJiraIssue",
-					rationaleCoverageCalculator.getCodeFilesWithNeighborsOfOtherType(KnowledgeType.DECISION));
-		} else {
-			IssueType jiraIssueType = JiraSchemeManager.createIssueType(issueType);
-			if (jiraIssueType != null) {
-				metrics.put("decisionsPerJiraIssue", rationaleCoverageCalculator
-						.getNumberOfDecisionKnowledgeElementsForJiraIssues(jiraIssueType, KnowledgeType.DECISION));
-				metrics.put("issuesPerJiraIssue", rationaleCoverageCalculator
-						.getNumberOfDecisionKnowledgeElementsForJiraIssues(jiraIssueType, KnowledgeType.ISSUE));
-				metrics.put("decisionDocumentedForSelectedJiraIssue", rationaleCoverageCalculator
-						.getJiraIssuesWithNeighborsOfOtherType(jiraIssueType, KnowledgeType.ISSUE));
-				metrics.put("issueDocumentedForSelectedJiraIssue", rationaleCoverageCalculator
-						.getJiraIssuesWithNeighborsOfOtherType(jiraIssueType, KnowledgeType.DECISION));
-			}
+		if (!sourceKnowledgeTypes.isEmpty()) {

```

Comment:
Please try to get rid of all the LinkedHashMaps in this class and pass the "rationaleCoverageCalculator" object directly.